src/net.cpp

```diff
@@ -2598,13 +2598,16 @@ ConnCounts CConnman::ConnectionCounts()
 {
     int num_in{0};
     int num_out{0};
+    bool tor_only{false};
     {
         LOCK(cs_vNodes);
+        if (!vNodes.empty()) tor_only = true;
         for (const auto& pnode : vNodes) {
             pnode->IsInboundConn() ? ++num_in : ++num_out;
+            if (!pnode->addr.IsTor()) tor_only = false;
```

Comment:
I didn't test if `IsTor()` behaves correctly when you have an inbound connection via a hidden service.

Reply:
> I didn't test if `IsTor()` behaves correctly when you have an inbound connection via a hidden service.

No, it doesn't. Going to rework his code.

Reply:
https://github.com/bitcoin/bitcoin/issues/8973

Reply:
If it's too tedious, I'm fine with the current change and just leaving correct handling of hidden service as a (well documented) TODO.

Reply:
The solution suggested by @laanwj submitted in https://github.com/bitcoin/bitcoin/pull/19991.